### PR TITLE
fix: enable Save button on Add Contact page for address input

### LINKED ISSUE #25889: [Bug]: Address Book - I cannot add a new account to the address book by pasting an address as the button is always disabled
### Describe the bug

I cannot add a new account to the address book by pasting an address. The button remains always disabled

### Expected behavior

_No response_

### Screenshots/Recordings

Screenshot from 2024-07-17 13-03-54

https://github.com/user-attachments/assets/c518b5f1-285f-4bc3-b2da-50b3b87d6bd7

### Steps to reproduce

1. Select Mainnet
2. Go to address book
3. Click Add account
4. Paste this address `0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045`
5. See button is always disabled, without any error
Note: you are able to add an account only if you add an ENS domain, that resolves to an address and the button is then disabled

### Error messages or log output

_No response_

### Detection stage

Develop

### Version

develop - not 12.0.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### LINKED ISSUE #25918: [Bug]: After entering a valid public account address into the Ethereum public address field in the Contacts list, the Save button remains greyed out.
### Describe the bug

After entering a valid public account address into the Ethereum public address field in Contacts/Address Book, the Save button remains greyed out, and the contact cannot be added. 

However, replacing the 0x address with an ENS address allows it to be added to this field and saved as a contact address without any issues.

### Expected behavior

You can add a contact by entering their public (0x) account address or ENS name without issue. 

### Screenshots/Recordings

https://github.com/user-attachments/assets/cad05a11-df35-4dcc-92c8-7e230c6b8f12

### Steps to reproduce

1. Go to Settings (on Mainnet)
2. Select Contacts
3. Click + Add contact
4. Enter a username and public account address with prefix 0x
5. Save button remains greyed out

### Error messages or log output

_No response_

### Detection stage

In beta

### Version

12.1.0

### Build type

Beta

### Browser

Firefox

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by mirceanis**
## **Description**

This patches an issue in `add-contact.component.js` where the disabled state of the `Save` button would disappear only after a successful ENS resolution, effectively preventing plain addresses to be entered.
I also added some extra unit tests to check for some of the cases that weren't covered before.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26155?quickstart=1)

## **Related issues**

fixes #25918
fixes #25889

## **Manual testing steps**

1. Go to Settings > Contacts > Add Contact
2. Enter a name and a valid Ethereum address
3. Observe the Save button getting enabled
4. Edit the address to make it invalid, observe the Save button getting disabled
5. Instead of an address, enter an ENS name and pick the resolved result
6. Observe the Save button getting enabled
7. Observe that the input is still editable (repeat 4.)

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 1]
```diff
@@ -150,7 +156,7 @@ export default class AddContact extends PureComponent {
                     domainName={addressBookEntryName ?? domainName}
                     onClick={() => {
                       this.setState({
-                        selectedAddress: resolvedAddress,
+                        input: resolvedAddress,
```
Comment (mirceanis) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:159]: This leaves the input field editable after an ENS resolved address has been picked

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.test.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ describe('AddContact component', () => {
     fireEvent.change(input, { target: { value: 'invalid address' } });
     setTimeout(() => {
       expect(getByText('Recipient address is invalid')).toBeInTheDocument();
-    }, 100);
+    }, 600);
```
Comment (mirceanis) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.test.js:50]: the debounce timeout is 500ms

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 3]
```diff
@@ -68,8 +68,10 @@ export default class AddContact extends PureComponent {
       isValidHexAddress(input, { mixedCaseUseChecksum: true });
     const validEnsAddress = isValidDomainName(input);
 
-    if (!IS_FLASK && !validEnsAddress && !valid) {
+    if (!validEnsAddress && !valid) {
```
Comment (NidhiKJha) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:71]: Do we not need it for FLASK?

Comment (mirceanis) [reply ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:71]: it doesn't matter if we are in FLASK or not, the validation is the same

**Review by NidhiKJha — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**